cmd/promtool/archive.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"os"
+)
+
+const FILE_PERM = 0644
+
+type archiveWriterConfig struct {
+	archiveName string
+}
+
+type archiveWriter interface {
```

Comment:
not a deal breaker, but maybe `archiveWriter` could be just `archiver`

Reply:
or can we use the io.WriteCloser?

Reply:
I think `archiver` is better! I'm not sure what do you mean by `io.WriteCloser`?

Reply:
I was thinking about this. The `write` method needs the argument of `filename`. If we want to implement the `Write(p []byte) (n int, err error)` method of `io.WriteCloser` instead, we will need to turn the argument of `filename` into the field of the implementation of `io.WriteCloser`, and every time before `Write()` is called, we need to reset the value of the `filename` so it can write to the corresponding file in the archive. It makes us need another method `SetHeader(filename string, size int64)` to this interface (currently `archiveWriter `) so we can't still just use `io.WriteCloser`. In this case, it will become something like:

```
type archiveWriter interface {
	io.WriteCloser
	SetHeader(filename string, size int64)
}
```
Another choice is that we embed only the `io.Closer` interface and let the `write()` method still handle the header setting part:
```
type archiveWriter interface {
	io.Closer
	write(fileName string, buf *bytes.Buffer) error
}
```
So what do you think?